### PR TITLE
allow memory limit set for gurobi 9.5.0 and higher

### PR DESCRIPTION
**Pull request by maxstb**

### CODE REVIEW

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 1]
```diff
@@ -97,13 +97,17 @@ mutable struct Env
     finalize_called::Bool
     attached_models::Int
 
-    function Env(; output_flag::Int = 1)
+    function Env(; output_flag::Int = 1, memory_limit::Number = 1.0e100)
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:100]:
```suggestion
    function Env(;
        output_flag::Int = 1,
        memory_limit::Union{Nothing,Real} = nothing,
    )
```

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 2]
```diff
@@ -97,13 +97,17 @@ mutable struct Env
...
         a = Ref{Ptr{Cvoid}}()
         ret = GRBemptyenv(a)
         env = new(a[], false, 0)
         _check_ret(env, ret)
         ret = GRBsetintparam(env.ptr_env, GRB_INT_PAR_OUTPUTFLAG, output_flag)
         _check_ret(env, ret)
+        if _GUROBI_VERSION >= v"9.5.0"
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:107]:
```suggestion
        if _GUROBI_VERSION >= v"9.5.0" && memory_limit !== nothing
```

**Review by odow — COMMENTED** [review]
Are there any other parameters that need to be set before the environment is started? If so, we should probably make a general fix, rather than these two explicitly parameters.

File: src/MOI_wrapper/MOI_wrapper.jl (hunk 1 above)
Comment (odow) [reply src/MOI_wrapper/MOI_wrapper.jl:100]: What about this option? Then we only set it if the user passes something, which guards against this default changing.

**Comment by maxstb** [comment]
> Are there any other parameters that need to be set before the environment is started? If so, we should probably make a general fix, rather than these two explicitly parameters.

I don't know about other parameters yet.

**Review by odow — APPROVED** [review]
This should do for now then. Don't worry about the CI failure, you don't have access to the license.

**Comment by maxstb** [comment]
Perfect, Thanks :)

**Pull request merged**